build-tools/config/webpack/webpack.conf.base.js

```diff
@@ -0,0 +1,24 @@
+const helpers = require('./webpack.helpers');
+const config = require('../config');
+
+module.exports = isDevelopment => {
```

Comment:
Should we change `isDevelopment` from a boolean to a string, so we can have more options?
I have some webpack projects where I have 3-4 different ways to build the project, all sharing the same util functions (with different params) and 1 or multiple base configs.

Having a 'buildtype' or something that can have multiple values, allows you to do switch/case or if/else in your _iife_ functions.

Reply:
Can you give an example of what these build types can be? The `isDevelopment` switches between development server and not a development server (so the opposite should be called "deployment" rather than "production", but unfortunately most webpack config out there doesn't follow that terminology). I feel like these "build types" would be a separate concern that can be changed independently of `isDevelopment`. But please correct me if I misunderstand

Reply:
Maybe this is also a broader question of: where does configuration go that is not static? As in, config that changes depending on what "type" of build you start or based on some kind of flag passed as command line argument. 
I think this is a slightly different concept than `isDevelopment`, because a project will always have a development build and a deployment build. These 2 builds shouldn't really differ in contents other then providing a development environment or not

Reply:
My suggestion would be to use variables in `config.js`, and switch between build types there

Reply:
In this example I have 4 'build types' / configs
- development
- production code
- production partials (similar to code, but with different loaders so the same source results in a different output... both code and partials extend the same base config, but are still different)
- production dev mode (that uses less plugins to make development faster, but still outputs everything in an output folder)

Currently it uses multiple extend levels, but in this new setup it would use if/else to add different kind of loaders, plugins, entry config, etc. So it would need the distinction between them.

Using config won't help, since that's also split up between dev and dist.

Reply:
Implemented in following commit:

https://github.com/hjeti/vue-skeleton/pull/82/commits/a059d401b5d1eb8118577e5d9e077880afd98a69

let me know if this is sufficient